File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool help(char a , char b)
    {
        if(a=='(' && b==')' || a=='{' && b=='}' || a=='[' && b==']')
        {
            return true;
        }
        else
        {
            return false;
        }
    }
    bool isValid(string str) {
        stack<char>s;
        for(auto key : str)
        {
            if(key == '(' || key == '{' || key == '[')
            {
                s.push(key);
            }
            else
            {
                if(s.empty()==true)
                {
                    return false;
                }
                else
                {
                    char ch = s.top();
                    bool check = help(ch,key);
                    if(check == true)
                    {
                        s.pop();
                    }
                    else
                    {
                        return false;
                    }
                }
            }
        }
        if(s.empty()==false)
            return false;
        return true;
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    
    public boolean check(Character ch , Character top){
        if(top == '[' && ch == ']' || top == '(' && ch == ')' || top == '{' && ch == '}' ){
            return true;
        }
        else{
            return false;
        }
    }
    
    public boolean isValid(String s) {
        
        
        Stack<Character> st = new Stack<>();
        
        for(int i=0;i<s.length();i++){
            char ch = s.charAt(i);
            
            
            if(st.empty() == true || (ch == '{' || ch == '('  || ch == '[' )){
                st.push(ch);
            }
            else{
                char top = st.peek();
                
                if( check(ch,top) == true){
                    st.pop();
                }
                else{
                    return false;
                }
            }
        }
        
        if(st.size() != 0){
            return false;
        }
        
        return true;
        
    }
}